### DISCUSSION TITLE
Amcrest HTTPS option support?

### DISCUSSION THREAD

**Discussion by stevenproctor** [koush/scrypted#807] [Ideas]
I was checking out this project as I am having issues with getting my Amcrest cameras to show in HomeAssistant and was attempting to use this as a bridge between the two.

I have a home lab with a reverse proxy setup using my domain and have wildcard LetsEncrypt certificates set up for that domain.

NGINX then proxies to the various Amcrest cameras via HTTPS to the cameras themselves.

This exposes signed certificates to various clients instead of the Amcrest self-signed certificates.

All this was done to prevent passwords to cameras from being insecurely transmitted over HTTP, even on a local network.

I saw in the plugin that it hard-codes the protocol to `"http://"`.

Is there a possibility of getting an option on the UI that asks for HTTP or HTTPS and then builds the connection strings based on that option's setting?

**Comment by koush** [comment]
Something this specific is not something I am likely to implement unless it is a pull request

**Reply by stevenproctor** [reply]
I am okay with looking into this and doing a PR if you are open to it.
